lambdas/check-reverification-identity/src/main/java/uk/gov/di/ipv/core/checkreverificationidentity/CheckReverificationIdentityHandler.java

```diff
@@ -0,0 +1,186 @@
+            LogHelper.attachGovukSigninJourneyIdToLogs(
+                    clientOAuthSession.getGovukSigninJourneyId());
+
+            var vcs =
+                    evcsService.getVerifiableCredentials(
+                            clientOAuthSession.getUserId(),
+                            clientOAuthSession.getEvcsAccessToken(),
+                            CURRENT);
+
+            var reverificationIdentityVot = getReverificationIdentityVot(vcs);
+            if (reverificationIdentityVot.isEmpty()) {
+                ipvSession.setFailureCode(NO_IDENTITY_AVAILABLE);
+                ipvSessionService.updateIpvSession(ipvSession);
+                return NOT_FOUND_RESPONSE;
+            }
+
+            if (GPG45.equals(reverificationIdentityVot.get().vot().getProfileType())) {
+                ipvSession.setTargetVot(reverificationIdentityVot.get().vot());
+                ipvSessionService.updateIpvSession(ipvSession);
+            }
+
+            return FOUND_RESPONSE;
+
+        } catch (HttpResponseExceptionWithErrorBody | EvcsServiceException e) {
+            LOGGER.error(LogHelper.buildErrorMessage(e.getErrorResponse()));
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH, e.getResponseCode(), e.getErrorResponse())
+                    .toObjectMap();
+        } catch (IpvSessionNotFoundException e) {
+            LOGGER.error(LogHelper.buildErrorMessage("IPV session not found", e));
+            return new JourneyErrorResponse(JOURNEY_ERROR_PATH, SC_NOT_FOUND, IPV_SESSION_NOT_FOUND)
+                    .toObjectMap();
+        } catch (CredentialParseException e) {
+            LOGGER.error(LogHelper.buildErrorMessage("Failed to parse credentials", e));
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH,
+                            SC_SERVER_ERROR,
+                            FAILED_TO_PARSE_SUCCESSFUL_VC_STORE_ITEMS)
+                    .toObjectMap();
+        } catch (ParseException e) {
+            LOGGER.error(LogHelper.buildErrorMessage("Failed to get VOT from operational VC", e));
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH, SC_SERVER_ERROR, FAILED_TO_PARSE_ISSUED_CREDENTIALS)
+                    .toObjectMap();
+        } catch (Exception e) {
+            LOGGER.error(LogHelper.buildErrorMessage("Unhandled lambda exception", e));
+            throw e;
+        }
+    }
+
+    private Optional<VotAndProfile> getReverificationIdentityVot(List<VerifiableCredential> vcs)
+            throws ParseException, HttpResponseExceptionWithErrorBody {
+        var gpg45Vcs = VcHelper.filterVCBasedOnProfileType(vcs, GPG45);
+        var gpg45Scores = gpg45ProfileEvaluator.buildScore(gpg45Vcs);
+        var gpg45VcsCorrelated = userIdentityService.areVcsCorrelated(gpg45Vcs);
+        var operationalVcs = VcHelper.filterVCBasedOnProfileType(vcs, OPERATIONAL_HMRC);
```

Comment:
Should we do the filtering in the `VotMatcher` - looks quite similar for both usages?

Reply:
I intentionally had them being passed in as we use the filtered VCs and the gpg45 scores elsewhere in the check-existing-identity lambda (for sending an audit event). I didn't want to have to do the filtering again, or send an audit event with details that weren't acutally used when checking the VOT.
Another option is we send that audit event (`IPV_GPG45_PROFILE_MATCHED`) in the new lambda but I figured it's not required and may be confusing.

Reply:
Ok sure - I think the overhead of filtering the list twice is pretty negligible, so I'd go for whichever looks neatest. The GPG45 scores are a bit more awkward (still pretty cheap to calculate, but worse than the filtering) :(

In terms of the audit event, I'm not sure! I could probably justify either way... If you want to retain the control in the calling code (which makes sense), I suppose you could have the method return a triple of Vot/GPG45Profile/GPG45Scores or something. Perhaps call it something like `ProfileMatchingResult` if it makes you feel better 😄

Reply:
Overall I quite like `ProfileMatchingResult matchProfile(List<Vot> vots, List<VerifiableCredential> vcs, List<ContraIndicator> cis)` as a good clean signature!